drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -319,7 +324,7 @@ local function lock_user_change_event_handler(driver, device, ib, response)
         lock_utils.code_deleted(device, cs)
       end
       lock_utils.lock_codes_event(device, {})
-      set_cota_credential(device)
+      if device:get_field(lock_utils.COTA_CRED) ~= nil then set_cota_credential(device) end
```

Comment:
Since the `false` case in `set_cota_credential` is commented as defensive/shouldn't happen, should that also be checked here before calling?

Reply:
Referring to this: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/69fa22cb492385b01a838318d79ac66ea9cf9011/drivers/SmartThings/matter-lock/src/init.lua#L39

Reply:
Looks like my defense was needed...I could have detected that we dont need to do anything here, but I instead added the defensive checks in `set_cota_credential`. I will remove the comment there, and add a docstring for the function.